File: Logic & Design/Visual Studio Community Edition/Structs/C++/StructFunction/StructFunction.cpp
/*
* Programmer:		Corey Jenkins
* Filename			StructFunciton.cpp
* Date:				September 27, 2022
* 
*/

#include <iostream>
#include <string>
using namespace std;

// Functions - three cows.
void cow1();
void cow2();
void cow3();

const enum COW_TYPE { dairy, pet, hide }; // Types of cows

struct Cow
{
	string name;
	int age;
	unsigned char purpose;
};

int main()
{
	cow1();
	cow2();
	cow3();

	system("pause");
	return 0;
}

void cow1()
{
	Cow MyCow1;
	MyCow1.name = "Betty";
	MyCow1.age = 5;
	MyCow1.purpose = dairy;

	cout << "This cow's name is " << MyCow1.name << endl;
	cout << "This cow is a type-" << (int)MyCow1.purpose << endl; // Type-0
	cout << "This cow is " << MyCow1.age << " years old." << endl;
}

void cow2()
{
	Cow MyCow2;
	MyCow2.name = "Betty";
	MyCow2.age = 7;
	MyCow2.purpose = pet;

	cout << "This cow's name is " << MyCow2.name << endl;
	cout << "This cow is a type-" << (int)MyCow2.purpose << endl; // Type-1
	cout << "This cow is " << MyCow2.age << " years old." << endl;

}

void cow3()
{
	Cow MyCow3;
	MyCow3.name = "Betty";
	MyCow3.age = 8;
	MyCow3.purpose = hide;

	cout << "This cow's name is " << MyCow3.name << endl;
	cout << "This cow is a type-" << (int)MyCow3.purpose << endl; // Type-2
	cout << "This cow is " << MyCow3.age << " years old." << endl;
}

File: Logic & Design/Visual Studio Community Edition/Structs/C#/StructFunction/StructFunction.cs
/*
 * Programmer:      Corey Jenkins
 * Filename:        StructFunction.cs
 * Date:            September 27, 2022
 * Purpose:         The use of an enumerator and structs to assign each cow's name, age and purpose.
*/
using System;

namespace StructFunction
{
    class StructFunction
    {
        enum CowType
        {
            dairy,
            hide,
            pet
        }
        struct cow
        {
            public string name;
            public int age;
        }
        static void Main(string[] args)
        {
            
            cow1();
            cow2();
            cow3();
            Console.ReadKey();
        }

        static void cow1()
        {
            cow MyCow1;
            MyCow1.name = "Betty";
            MyCow1.age = 5;
            CowType myCow = CowType.dairy;

            Console.WriteLine("This cow's name is " + MyCow1.name + ".");
            Console.WriteLine(MyCow1.name + " is " + MyCow1.age + "Years old.");
            Console.WriteLine(MyCow1.name + " is a cow for " + myCow + ".");
        }

        static void cow2()
        {
            cow MyCow1;
            MyCow1.name = "Abigail";
            MyCow1.age = 7;
            CowType myCow = CowType.hide;

            Console.WriteLine("This cow's name is " + MyCow1.name + ".");
            Console.WriteLine(MyCow1.name + "is " + MyCow1.age + " Years old.");
            Console.WriteLine(MyCow1.name + " is a cow for " + myCow + ".");

        }

        static void cow3()
        {
            cow MyCow1;
            MyCow1.name = "The Cow King";
            MyCow1.age = 9;
            CowType myCow = CowType.pet;

            Console.WriteLine("This cow's name is " + MyCow1.name + ".");
            Console.WriteLine(MyCow1.name + " is " + MyCow1.age + " Years old.");
            Console.WriteLine(MyCow1.name + " is a cow for a " + myCow + ".");
        }
    }
}
